functional/churn/playlist.xml

```diff
@@ -0,0 +1,88 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../TKG/resources/playlist.xsd">
+	<test>
+		<testCaseName>churn_1m_allGCs</testCaseName>
+		<command> 
+            export OTOOL_JDK_VERSION="$(JDK_VERSION)" ; \
+            export JREJDK="jdk" ; \
+            export OTOOL_garbageCollector="ALL" ; \
+            export DURATION="60" ; \
+            export TMPRESULTS="$(Q)$(REPORTDIR)$(D)report$(Q)"; \
+            cd $TMPRESULTS ; \
+              bash "$(TEST_ROOT)$(D)functional$(D)churn$(D)churn$(D)run.sh"; \
+            $(TEST_STATUS)
+		</command>
+		<levels>
+			<level>dev</level>
+		</levels>
+		<platformRequirementsList>
+			<platformRequirements>os.linux</platformRequirements>
+			<platformRequirements>os.win</platformRequirements>
+			<platformRequirements>os.osx</platformRequirements>
+		</platformRequirementsList>
+		<groups>
+			<group>system</group>
+		</groups>
+	</test>
+	<test>
+		<testCaseName>churn_5h_allGCs</testCaseName>
+		<command> 
+            export OTOOL_JDK_VERSION="$(JDK_VERSION)" ; \
+            export JREJDK="jdk" ; \
+            export OTOOL_garbageCollector="ALL" ; \
+            export DURATION="18000" ; \
+            export TMPRESULTS="$(Q)$(REPORTDIR)$(D)report$(Q)"; \
+            cd $TMPRESULTS ; \
+              bash "$(TEST_ROOT)$(D)functional$(D)churn$(D)churn$(D)run.sh"; \
+            $(TEST_STATUS)
+		</command>
+		<levels>
+			<level>dev</level>
+		</levels>
+		<platformRequirementsList>
+			<platformRequirements>os.linux</platformRequirements>
+			<platformRequirements>os.win</platformRequirements>
+			<platformRequirements>os.osx</platformRequirements>
+		</platformRequirementsList>
+		<groups>
+			<group>system</group>
+		</groups>
+	</test>
+	<test>
+		<testCaseName>churn_custom</testCaseName>
+		<command> 
+            export OTOOL_JDK_VERSION="$(JDK_VERSION)" ; \
+            export JREJDK="jdk" ; \
+            export OTOOL_garbageCollector="${CHURN_GCS}" ; \
```

Comment:
```suggestion
            export OTOOL_garbageCollector="${APPLICATION_OPTIONS}" ; \
```

Reply:
I do not understand this requirements:
```
- export OTOOL_garbageCollector="${CHURN_GCS}" ; \
+export OTOOL_garbageCollector="${APPLICATION_OPTIONS}" ; \
...
- export DURATION="${CHURN_DURATION}" ; \
+ export DURATION="300" ; \
```
The APPLICATION_OPTIONS is something what is custom in aqavit? Isn't it some more general setup?
The "zgc" or "shenandaoh" can be hardly considered as generic.

To remove custom duration s moreover killing the purpise of the changeset and the suite. If you are developing custom GC or porting JDK to new paltform, you want to run churn on each GC you care about (usually default or the single custom one) for several hours. 300s is useles.  Generally any hardcoded value in the "custom" setupo is (IMO) bad.

Can we stay with originals pelase? Or maybe an APPLICATION_OPTIONS contain key=value pairs? (but as GC can be space separated list of GCs, it sounds bad). Or maybe APPLICATION_OPTIONS_GC and APPLICATION_OPTIONS_DURATION ?

Reply:
Seeing your touch to readme - is there eal fear, that some ccrazy eprson will start 20days long churn run? If so, then I would lke to still keep the duration adjsutable, and only eg limit it from top.

if such logic would be desirable, then also the  churn_1m_allGCs and churn_5h_allGCs may actually take GC as parameter. And fallback to "ALL" if nothing provided. wdyt?

Reply:
Please see that I have asked that this churn_custom target is disabled (using the `<disables>` block I suggested), in that way it does not get run during our automated runs.  

Also in our world, no one is allowed to go onto test machines and set CHURN_DURATION and CHURN_GCS environment variables, and we would not want them to do that (for lots of reasons, not the least of which is transparency of machine config).

This leaves me to the point to look at this test target as a useful thing when someone wants to run a Grinder to test a particular GC for a short duration, which is why I suggest leveraging the existing APPLICATION_OPTIONS mechanism for pinpointed testing of a particular GC failure.

All that to be said, feel free to not take those review comments and apply them.  

> if such logic would be desirable, then also the churn_1m_allGCs and churn_5h_allGCs may actually take GC as parameter. And fallback to "ALL" if nothing provided. wdyt?

Yes, the other option is to remove the churn_custom target altogether from the playlist.xml file in this PR.  That would be the expedient way to get this PR through.

Could consider then leveraging APPLICATION_OPTIONS to pass a subset of GCs and if its blank, expect the churn test suite to default to all.